diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ee01e7c..436b07a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -100,16 +100,19 @@ endif()
 # Tests
 #============================================================================
 if(WITH_TESTS)
-	set(TESTPROG "${CMAKE_CURRENT_BINARY_DIR}/run_testcase")
+	set(TESTPROG "${CMAKE_BINARY_DIR}/bin/run_testcase")
 	set(TESTCASES "${PROJECT_SOURCE_DIR}/testcases.docopt")
 	add_executable(run_testcase run_testcase.cpp)
-	target_link_libraries(run_testcase docopt)
+	target_link_libraries(run_testcase docopt_s)
 	configure_file(
 			"${PROJECT_SOURCE_DIR}/run_tests.py"
 			"${CMAKE_CURRENT_BINARY_DIR}/run_tests"
 			ESCAPE_QUOTES
 	)
-	add_test("Testcases docopt" ${TESTPROG})
+	add_test(
+		NAME "Testcases_docopt"
+		COMMAND python "${CMAKE_CURRENT_BINARY_DIR}/run_tests"
+	)
 endif()
 
 #============================================================================
diff --git a/src/run_tests.py b/src/run_tests.py
index ef723b4..39d0969 100755
--- a/src/run_tests.py
+++ b/src/run_tests.py
@@ -56,17 +56,17 @@ for _, doc, cases in parse_test(tests):
 
 		failures += 1
 
-		print "="*40
-		print doc
-		print ':'*20
-		print prog, argv
-		print '-'*20
+		print("="*40)
+		print(doc)
+		print(':'*20)
+		print(prog, argv)
+		print('-'*20)
 		if out:
-			print out
-		print error
+			print(out)
+		print(error)
 
 if failures:
-	print "%d failures" % failures
+	print("%d failures" % failures)
 	sys.exit(1)
 else:
-	print "PASS (%d)" % passes
+	print("PASS (%d)" % passes)
